ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +1052,242 @@ def test_server_option_with_unreachable_ad(self):
             self.master.run_command(['rm', '-f', ad_zone_file])
             tasks.configure_dns_for_trust(self.master, self.ad)
             self.remove_trust(self.ad)
+
+
+class TestNonPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+
+    @classmethod
+    def install(cls, mh):
+        if not cls.master.transport.file_exists('/usr/bin/rpcclient'):
+            raise pytest.skip("Package samba-client not available "
+                              "on {}".format(cls.master.hostname))
+        super(TestNonPosixAutoPrivateGroup, cls).install(mh)
+        cls.ad = cls.ads[0]
+        cls.ad_domain = cls.ad.domain.name
+        tasks.install_adtrust(cls.master)
+        tasks.sync_time(cls.master, cls.ad)
+        cls.existing_rangename = cls.ad_domain.upper() + "_id_range"
+
+    def test_add_nonposix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust'])
+
+    def test_modify_auto_private_groups_value_default_trusted_range(self):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as true
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(self.existing_rangename, "true")
+        result = self.master.run_command(["id", nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        gid_reg = (
+            r"gid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
```

Comment:
In this test, you check the output for a user which has an existing gid defined in AD. It would be interesting to check with a user which doesn't have any gid defined in AD or for a user which has a gid set but no corresponding group in AD. The same comment applies to all the possible settings for auto-private-group.
It means that you need to modify the ansible playbook that is creating the AD users (see https://github.com/freeipa/freeipa-pr-ci/tree/master/ansible/roles/windows/ipa-ad-data) and also create a user without gid, and a user with gid but no corresponding group.
Same thing for non-posix users: a user with a primaryGroupID that is not the default one.

Reply:
Please address my previous comment: the test should ensure that gid=uid. You can reuse the method get_user_id here.